lib/cli/src/commands/run/mod.rs

```diff
@@ -717,7 +717,7 @@ impl ExecutableTarget {
                 let module = unsafe { Module::deserialize_from_file(&engine, path)? };
                 let module_hash = {
                     let wasm = std::fs::read(path)?;
-                    ModuleHash::hash(wasm)
+                    ModuleHash::xxhash(wasm)
```

Comment:
I just realized that this is completely wrong, `let wasm = std::fs::read(path)?` is not a wasm file, but a precompiled artifact.
Thus, we shall not use that as the hash in any way.

Reply:
Issue: https://github.com/wasmerio/wasmer/issues/4632

Reply:
Since we decided to track this bug in a separate issue and solve it later on, I'll resolve this comment.